addons/sale_mrp/tests/test_sale_mrp_flow.py

```diff
@@ -383,3 +384,100 @@ def test_01_sale_mrp_delivery_kit(self):
+        self.partner.property_account_receivable_id = self.account
+        self.category.property_account_income_categ_id = self.account
+        self.category.property_account_expense_categ_id = account_expense
+        self.category.property_stock_account_input_categ_id = self.account
+        self.category.property_stock_account_output_categ_id = account_output
+        self.category.property_stock_valuation_account_id = self.account
+        self.category.property_stock_journal = self.env['account.journal'].create({'name': 'Stock journal', 'type': 'sale', 'code': 'STK00'})
+        self.finished_product = self.Product.create({
+                'name': 'Finished product',
+                'type': 'product',
+                'uom_id': self.uom_unit.id,
+                'invoice_policy': 'delivery',
+                'categ_id': self.category.id})
+        self.component1 = self.Product.create({
+                'name': 'Component 1',
+                'type': 'product',
+                'uom_id': self.uom_unit.id,
+                'categ_id': self.category.id,
+                'standard_price': 20})
+        self.component2 = self.Product.create({
+                'name': 'Component 2',
+                'type': 'product',
+                'uom_id': self.uom_unit.id,
+                'categ_id': self.category.id,
+                'standard_price': 10})
+        self.bom = self.MrpBom.create({
+                'product_tmpl_id': self.finished_product.product_tmpl_id.id,
+                'product_qty': 1.0,
+                'type': 'phantom',
+                'product_efficiency': 1.0,
+                'product_uom': self.uom_unit.id})
+        self.MrpBomLine.create({
+                'product_id': self.component1.id,
+                'product_qty': 2.0,
+                'bom_id': self.bom.id,
+                'product_uom': self.uom_unit.id})
+        self.MrpBomLine.create({
+                'product_id': self.component2.id,
+                'product_qty': 1.0,
+                'bom_id': self.bom.id,
+                'product_uom': self.uom_unit.id})
+
+        # Create a SO for a specific partner for three units of the finished product
+        so_vals = {
+            'partner_id': self.partner.id,
+            'partner_invoice_id': self.partner.id,
+            'partner_shipping_id': self.partner.id,
+            'order_line': [(0, 0, {'name': self.finished_product.name, 'product_id': self.finished_product.id, 'product_uom_qty': 3, 'product_uom': self.finished_product.uom_id.id, 'price_unit': self.finished_product.list_price})],
+            'pricelist_id': self.env.ref('product.list0').id,
+            'company_id': self.company.id,
+        }
+        self.so = self.SaleOrder.create(so_vals)
+        # Validate the SO
+        self.so.action_confirm()
+        # Deliver the three finished products
+        pick = self.so.picking_ids
```

Comment:
Basic `assert` could be added: check what are the products on the picking (it should be `self.component1` and `self.component2`). e.g.:
`self.assertEqual(pick.move_lines.mapped('product_id'), self.component1 | self.component2)`